libpromises/conversion.c

```diff
@@ -509,7 +509,7 @@ bool DoubleFromString(const char *s, double *value_out)
 
     remainder[0] = '\0';
 
-    sscanf(s, "%lf%c%s", &a, &c, remainder);
+    sscanf(s, "%lf%c%4095s", &a, &c, remainder);
```

Comment:
Err, not cherry-picking after all since you've already changed one occurence of this kind. Please do the same here.